### PR TITLE
GEODE-5694 Do not create or publish zip distributions

### PR DESCRIPTION
**Pull request by smgoller**
Zip archive is removed from publication list.
Zip distribution tasks are disabled by default, but may be explicitly
asked for on the command line.
Combined taskGraph tasks in the same Gradle file into one call.

Signed-off-by: Sean Goller <sgoller@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by metatype** [comment]
I think the geode-examples build is currently pulling down the zip distribution.  Can you also update [1] to use the tarball instead?

[1] https://github.com/apache/geode-examples/blob/master/build.gradle#L49

**Comment by metatype** [comment]
@pivotal-jbarrett Nice! 👍

File: geode-assembly/build.gradle [hunk 1]
```diff
@@ -54,6 +55,7 @@ configurations.all {
 }
 
 gradle.taskGraph.whenReady( { graph ->
+  tasks.install.enabled = false
```
[the remark this answers is not in the document]
Comment (rhoughton-pivot) [reply geode-assembly/build.gradle]: The task comes from the `maven` plugin, and has been replaced by `maven-publish` which uses the `publishToMavenLocal` task, which better handles disabled artifacts.

File: geode-assembly/build.gradle [hunk 2]
```diff
@@ -68,6 +70,29 @@ gradle.taskGraph.whenReady( { graph ->
     distTar.dependsOn(jarTasks)
     distZip.dependsOn(jarTasks)
   }
+  srcDistZip.enabled = false
+  distZip.enabled = false
+
+  project.getGradle().startParameter.getTaskNames().each { givenTask ->
```
[the remark this answers is not in the document]
Comment (rhoughton-pivot) [reply geode-assembly/build.gradle]: This logic is to allow someone to explicitly ask for a `distZip` or `distSrcZip`, but to not spend time on them in a normal `build` task

[the remark this answers is not in the document]
Comment (metatype) [reply geode-assembly/build.gradle]: Let's just disable entirely and keep it simple :-)

**Pull request merged**